### PR TITLE
Connects Falkonry/tracker#936 Batch changes

### PR DESCRIPTION
**Pull request by dilpeshjain**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,10 +22,14 @@ Falkonry C# Client to access [Falkonry Condition Prediction](falkonry.com) APIs
...
-    * Add wide input data (json format) to single entity Datastream
-    * Add wide input data (csv format) to multi entity Datastream 
+	* Add historical narrow input data (json format) to multi entity Datastream (Used for model revision)
+	* Add historical narrow input data (csv format) single entity to Datastream (Used for model revision)
+    * Add historical wide input data (json format) to single entity Datastream (Used for model revision)
+    * Add historical wide input data (csv format) to multi entity Datastream (Used for model revision)
+	* Add historical narrow input data (json format) to multi entity Batch Datastream (Used for model revision)
+	* Add historical narrow input data (csv format) single entity to BatchDatastream (Used for model revision)
+    * Add historical wide input data (json format) to single entity Batch Datastream (Used for model revision)
+    * Add historical wide input data (csv format) to multi entity Batch Datastream (Used for model revision)
```
Comment (aniket-amrutkar) [inline README.md]: Add one more test case for datastream creation with batch Identifier

File: README.md [hunk 2]
```diff
@@ -512,8 +516,97 @@ Falkonry C# Client to access [Falkonry Condition Prediction](falkonry.com) APIs
...
+    options.Add("timeIdentifier", "time");
+    options.Add("timeZone", "GMT");
+    options.Add("timeFormat", "YYYY-MM-DD HH:mm:ss");
+    options.Add("signalIdentifier", "signal");
+    options.Add("valueIdentifier", "value");
+	options.Add("batchIdentifier", "batch");
+    var inputstatus = _falkonry.AddInput(datastream.Id, data, options);
+```
+
+#### Add historical narrow input data (csv format) multi entity to Batch Datastream (Used for model revision)
```
Comment (aniket-amrutkar) [inline README.md]: remove italic formatting

File: README.md [hunk 3]
```diff
@@ -684,6 +777,27 @@ Falkonry C# Client to access [Falkonry Condition Prediction](falkonry.com) APIs
...
+    using falkonry_csharp_client;
+    using falkonry_csharp_client.helper.models;
+
+	SortedDictionary<string, string> options = new SortedDictionary<string, string>();
+	
+
+    string token = "Add your token here";   
+    SortedDictionary<string, string> options = new SortedDictionary<string, string>();
+    Falkonry falkonry = new Falkonry("http://localhost:8080", token);
+    String data = "{\"time\" : \"2011-03-26T12:00:00.000Z\", \"end\" : \"2012-06-01T00:00:00.000Z\", \"Health\" : \"Normal\", \"Batch\" : \"batch1\"}";
```
Comment (aniket-amrutkar) [inline README.md]: Why are you sending start time and end time for facts batch ingestion?

Comment (aniket-amrutkar) [reply README.md]: Also add test case for get facts for batch datastream

File: src/helper/models/Signal.cs [hunk 4]
```diff
@@ -2,30 +2,13 @@
 {
     public class Signal
     {
-        public string TagIdentifier
```
Comment (aniket-amrutkar) [inline src/helper/models/Signal.cs]: Put this code back as we are still supporting old narrow format data

File: test/FalkonryTests.cs [hunk 5]
```diff
@@ -74,8 +75,7 @@ public void createDatastreamNarrowFormatSingleEntity()
             var Field = new Field();
             var Signal = new Signal();
             Signal.ValueIdentifier = "value";
-            Signal.TagIdentifier = "tag";
```
Comment (aniket-amrutkar) [inline test/FalkonryTests.cs]: Same here. Keep the old code as it is

**Comment by phagunbaya** [comment]
@aniket-amrutkar @dilpeshjain Please plan to merge this.

File: src/helper/models/Signal.cs (hunk 4 above)
Comment (aniket-amrutkar) [reply src/helper/models/Signal.cs]: @dilpeshjain Did you address this ?

File: test/FalkonryTests.cs (hunk 5 above)
Comment (aniket-amrutkar) [reply test/FalkonryTests.cs]: @dilpeshjain Did you address this ?

File: README.md (hunk 2 above)
Comment (dilpeshjain) [reply README.md]: It is not Italic in the original file, somehow it is showing here.

**Pull request merged**